splito/lohi/_lo.py

```diff
@@ -0,0 +1,160 @@
+from rdkit import Chem, DataStructs
+from rdkit.Chem import AllChem
+import numpy as np
+from tqdm import tqdm
+
+from ._utils import get_similar_mols
+
+class LoSplitter():
+    def __init__(self, threshold=0.4, min_cluster_size=5, max_clusters=50, std_threshold=0.60):
+        """
+        Parameters:
+            threshold -- molecules with a similarity higher than this value are considered similar.
+            min_cluster_size -- number of molecules per cluster.
+            max_clusters -- maximum number of selected clusters. The remaining molecules go to the training set.
+            std_threshold -- Lower bound of the acceptable standard deviation for a cluster. It should be greater than measurement noise.
+                            If you're using ChEMBL-like data, set it to 0.60 for logKi and 0.70 for logIC50.
+                            Set it lower if you have a high-quality dataset.
+        """
+        self.threshold = threshold
+        self.min_cluster_size = min_cluster_size
+        self.max_clusters = max_clusters
+        self.std_threshold = std_threshold
+
+    def split(self, smiles, values, verbose=1):
+        """
+        Split the dataset of smiles and their continuous values.
+
+        Parameters:
+            smiles -- list of smiles.
+            values -- list of their continuous activity values.
+            verbose -- set to 0 to turn off progressbar.
+
+        Returns:
+            train_idx -- list of train indices.
+            clusters_idx -- list of lists of cluster indices.
+        """
+        if not isinstance(smiles, np.ndarray):
+            smiles = np.array(smiles)
+        if smiles.shape != np.unique(smiles).shape:
+            raise ValueError("Remove duplicates from your smiles")
```

Comment:
I would remove this. 

I agree that in most cases this makes total sense, but I would nevertheless leave it to the user. For example, someone might use resampling strategies, leading to intentional duplicates!

Reply:
Done. It required some changes in the algorithm.